ksqldb-engine/src/main/java/io/confluent/ksql/security/KsqlProvidedAccessValidator.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.security;
+
+import static java.util.Objects.requireNonNull;
+
+import com.google.common.collect.ImmutableList;
+import org.apache.kafka.common.acl.AclOperation;
+
+/**
+ * An implementation of {@link KsqlAccessValidator} that provides authorization checks
+ * from a external authorization provider.
+ */
+public class KsqlProvidedAccessValidator implements KsqlAccessValidator {
```

Comment:
nit: why not just name it `KsqlAccessValidatorImpl`? `KsqlProvidedAccessValidator` sounds a bit weird.

Reply:
There are 3 classes that extend from this:
```
KsqlProvidedAccessValidator
KsqlCacheAccessValidator
KsqlBackendAccessValidator
```

I used `Provided...` because it is a validator  provided by the user on the `ksql-server.properties`. I wasn't sure if use this os `KsqlExternalAccessValidator`. Any ideas?